daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -695,12 +695,12 @@ abstract class TestCase(testCaseXML: NodeSeq, val parent: DFDLTestSuite) {
   }
 
   def getRootNamespaceString() = {
-    if (optExpectedOrInputInfoset.isDefined)
+    if (this.rootNSAttrib != "")
+      rootNSAttrib
```

Comment:
I wonder if rootNSAttrib wants to be an Option? So that `rootNS=""` is `Some("")` explicitly means no-namespace, and if rootNS is missing then it's None and we just search for any element with the name `root`, and errors if there are ambiguities?